### PR TITLE
add script for running input files through classifiers

### PR DESCRIPTION
**Pull request by lisac**
## summary of changes

Introduces a script that allows for running a single input file (representing conditions and their classifications) through multiple classifiers, with outputs representing the predictions from those classifiers. The intent is to gather outputs for comparing and tracking behavior of the classifiers.

closes https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/775

### CODE REVIEW

File: src/python_src/util/data/simulations/run_simulations.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    with open(os.path.join(SIMULATIONS_DIR, "outputs", filename), "w") as f:
+        f.writelines(classification_report)
+    return filename
+
+
+def _write_predictions_to_file(
+    conditions_to_test: List[str], expected_classifications: List[str], classifier: BaseClassifierForSimulation
+) -> str:
+    filename = f"{c.name.replace(" ", "_")}_{TIMESTAMP}.csv"
+
```
[the remark this answers is not in the document]
Comment (lisac) [reply src/python_src/util/data/simulations/run_simulations.py:42]: This is a bug. 
Yes, `c` is meant to be a Classifier object.
Oddly, it's not causing an issue. The intended behavior is to create a filename such as `respiratory_classifier_2025-06-16_11-37-23.csv`, and i've double-checked that this is happening. 

I wonder if - conveniently, in this case -  sthis function is aware of the `c` in `for c in classifiers:` in `__main__`. 

TODOs, I'll update this PR to: 
- add unit tests (all over)
- update this particular method to not rely on having access to `c`.
    - `filename = f"{c.name.replace(" ", "_")}_{TIMESTAMP}.csv"` --> `filename = f"{classifier.name.replace(" ", "_")}_{TIMESTAMP}.csv"`
    - why: more readable, and avoid reliance on an obscure behavior.

Thank you for flagging this, @sdbars.

File: src/python_src/util/data/simulations/run_simulations.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        c.make_predictions(conditions_to_test)
+
+        print(c.predictions)
+        predictions_file = _write_predictions_to_file(conditions_to_test, expected_classifications, c)
+
+        labels = list(set(expected_classifications + c.predictions))
+        labels.sort()
+        computed_scores = classification_report(expected_classifications, c.predictions, target_names=labels, zero_division=1)
+
```
Comment (sdbars) [inline src/python_src/util/data/simulations/run_simulations.py:136]: We’re currently passing only target_names. When labels is omitted as a parameter, classification_report infers its own label list from the data and then checks that its length / order matches target_names. The [doc](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html) shows labels and target_names look to the same thing:  array-like of shape (n_labels,), default=None. That might be okay for now because the lists happen to be identical, but might complain if future datasets add or reorder labels.

File: src/python_src/util/data/simulations/run_simulations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lisac) [reply src/python_src/util/data/simulations/run_simulations.py:41]: added unit tests in 2f02500 and corrected the variable name in 7afefe0

File: src/python_src/util/data/simulations/run_simulations.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+SIMULATIONS_DIR = "src/python_src/util/data/simulations/"
+INPUT_FILE = "inputs_mini.csv"
+TIMESTAMP = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+
+
+def _write_scores_to_file(classification_report: str, file_prefix: str) -> str:
+    filename = f"{file_prefix}_{TIMESTAMP}_scores.txt"
+
+    with open(os.path.join(SIMULATIONS_DIR, "outputs", filename), "w") as f:
+        f.writelines(classification_report)
```
[the remark this answers is not in the document]
Comment (lisac) [reply src/python_src/util/data/simulations/run_simulations.py]: switched to `write()` in dcb4bd8, and I hope not to overlook this again! Thank you for the tip.

File: src/python_src/util/data/simulations/run_simulations.py (hunk 2 above)
Comment (lisac) [reply src/python_src/util/data/simulations/run_simulations.py:134]: great flag. I've added `labels` as an explicit parameter for `classification_report()`. 

I'll also reach out to the AI team to ask for a review.

File: src/python_src/util/data/simulations/run_simulations.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+        print(c.predictions)
+        predictions_file = _write_predictions_to_file(conditions_to_test, expected_classifications, c)
+
+        labels = list(set(expected_classifications + c.predictions))
+        labels.sort()
+
+        metrics_file = _write_metrics_to_file(
+            classification_report(expected_classifications, c.predictions, 
+                labels=labels, target_names=labels, zero_division=1),
```
Comment (lisac) [inline src/python_src/util/data/simulations/run_simulations.py:137]: @emilymechva @nelabdiel  am i calling  `classification_report()` correctly?

as sample values of the parameters:
`expected_classifications` and `c.predictions` are both lists of strings, for example `['blue', 'red', 'blue']`, `['blue', 'red', 'red']`; `labels` is the union of those two lists, de-duplicated and sorted (eg `['blue','green']`).   I set `zero_division=1` because the console told me to.

**Comment by lisac** [comment]
thanks for the reviews @ZachMurray @sdbars . I've made some corrections - good! - although managed to trigger something that made me lose the "PR Approved" status. would you mind taking another look? thanks!

File: src/python_src/util/data/simulations/run_simulations.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+        for i in range(len(conditions_to_test)):
+            row_to_write = f"{conditions_to_test[i]},{expected_classifications[i]}"
+
+            for c in classifiers:
+                row_to_write += f",{c.predictions[i]},{c.predictions[i] == expected_classifications[i]}"
+
+            f.write(f"{row_to_write}\n")
+    return filename
+
```
Comment (sdbars) [inline src/python_src/util/data/simulations/run_simulations.py:97]: This code is fine so long as data integrity is ensured. Here's where I see a problem:

```
conditions_to_test = ["acne", "some, complex condition"]
expected_classifications = ["9016", "9012"]
```

Running `f"{conditions_to_test[i]},{expected_classifications[i]}"`

Will produce the following output:

```
acne,9016
some, complex condition,9012
```

I think this is fine for now. It might need refactoring in the future. Maybe using python's [csv writer](https://docs.python.org/3/library/csv.html)?

**Review by ZachMurray — APPROVED** [review]
Everything looks good! 
There's always improvements that can be made, 
Which we can do in the future and on the side, if need be.

File: src/python_src/util/data/simulations/run_simulations.py (hunk 5 above)
Comment (lisac) [reply src/python_src/util/data/simulations/run_simulations.py:97]: On board with `csv writer`. i had updated `_write_predictions_to_file()` in my code to use that and overlooked this function. 🤦‍♀️    Will address this and the other comments in a follow-up PR this week.
